### ISSUE TITLE
Strategy for daily model and sfc type model implementation in OGGM

### ISSUE THREAD

**Issue by fmaussion** [OGGM/oggm#1703]
@bearecinos after our meeting and some thinking with @lilianschuster, I'd suggest the following:

If you can solve most of the failing tests on mb-sandbox in a reasonable time, it's worth spending the effort to have a benchmark we can work with. However, @lilianschuster said that there is also quite some boiler-plate code that might hinder development itself, so if this gets too complicated maybe drop this step altogether.

Regardless of the outcome of the above, perhaps the easiest would still be to start from scratch on OGGM core itself, by:
- starting a new module called massbalance_sandbox.py in oggm/core
- creating a first class called `DailyTIModel` which mirrors [MonthlyTIModel](https://github.com/OGGM/oggm/blob/9faee85f6b8f7c16c20f1037735f73d05d61080f/oggm/core/massbalance.py#L317) but uses @lilianschuster's code to deal with daily inputs
- `DailyTIModel` can be tested the same way as `MonthlyTIModel` and it can be quite shallow, for example by copiing some of the tests in https://github.com/OGGM/oggm/blob/9faee85f6b8f7c16c20f1037735f73d05d61080f/oggm/tests/test_prepro.py#L1410 
- Surface type distinction will come after that.

Thoughts?

**Comment by bearecinos** [comment]
Yep sounds good ...  happy to start the script and the class and ask input when needed. The aim first would be to re-create that `test_prepro.py::test_mb_calibration_from_scalar_mb` with @lilianschuster's classes and go from there... to put together the new `massbalance_sandbox.py` script.

**Comment by bearecinos** [comment]
@fmaussion @lilianschuster 
I've started working on this and the first thing that I notice is that before the `DailyTIModel` class we might need another Superclass e.g. [`class MassBalanceModel(object, metaclass=SuperclassMeta):`](https://github.com/OGGM/oggm/blob/master/oggm/core/massbalance.py#L35) at the top? 

Lili's code has a different way of defining that though.. She has a [TIModel_Parent class](https://github.com/OGGM/massbalance-sandbox/blob/master/MBsandbox/mbmod_daily_oneflowline.py#L869) which does not look quite the same? The parent class deals with chosen different MB types within Lili's code (e.g. smb with surface type distinction etc)
The question is perhaps more for @lilianschuster:
The code that we want to keep and integrate is it both? the: [`TIModel_Parent(MassBalanceModel)`](https://github.com/OGGM/massbalance-sandbox/blob/master/MBsandbox/mbmod_daily_oneflowline.py#L869C4-L869C40) and [`TIModel(TIModel_Parent):`](https://github.com/OGGM/massbalance-sandbox/blob/master/MBsandbox/mbmod_daily_oneflowline.py#L2071) or just the `TIModel(TIModel_Parent):`
